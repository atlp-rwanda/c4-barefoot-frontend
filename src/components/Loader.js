import React from 'react';
import {Modal, Fade, Typography, CircularProgress} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme) =>({
    modal:{
        display: 'Flex',
        alignItems: 'center',
        justifyContent: 'center'
    },
    loader:{
        backgroundColor: theme.palette.background.paper,
        boxShadow: theme.shadows[5],
        outline: 'none',
        padding: theme.spacing(2, 4, 3),
        textAlign: 'center'
    }

}));

function Loader (props) {
    const { t, i18n } = useTranslation();

    const classes = useStyles();
    return (
        <React.Fragment>
            <Modal
            className={classes.modal}
            open={props.open}
            closeAfterTransition
            >
                <Fade in={props.open}>
                    <div className={classes.loader}>
                        <CircularProgress/>
                        <Typography>{t("Processing ...")}</Typography>
                    </div>
                </Fade>
            </Modal>
        </React.Fragment>
    )
}

export default Loader;