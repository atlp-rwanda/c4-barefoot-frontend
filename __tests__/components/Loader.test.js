import React from 'react';
import Loader from '../../src/components/Loader';
import {Fade, CircularProgress,Typography, Modal} from '@material-ui/core';
import {shallow, mount} from 'enzyme';
import toJson from "enzyme-to-json";
describe('<Loader />', () =>{
    let wrapper;
    it('should be rendered without errors', ()=>{
        const props ={ open:true}
        wrapper = shallow(<Loader {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    it('should open', () =>{
        expect(wrapper.find(Fade).props().children.props.children[1].props.children).toEqual("Processing ...");
        expect(wrapper.find(Fade).length).toBe(1);
        expect(wrapper.find(Modal).length).toBe(1);
        expect(wrapper.find(Modal).props().open).toBe(true);
        expect(wrapper.find(Fade).props().in).toBe(true);
        
    })
    it('should close', () =>{
        const props ={ open:false}
        wrapper = shallow(<Loader {...props}/>);
        expect(wrapper.find(Fade).props().in).toBe(false);
        expect(wrapper.find(Modal).props().open).toBe(false);

    });
    it('should have 1 circular Progress', () =>{
        expect(wrapper.find(CircularProgress).length).toBe(1);
    })
    it('should have 1 Typography', () =>{
        expect(wrapper.find(Typography).length).toBe(1);
    })
    it('should have the Typography of Processing ...', () =>{
        expect(wrapper.find(Fade).props().children.props.children[1].props.children).toEqual('Processing ...');
    })
    it('should use mount as well', () =>{
        const props={open: true}
        wrapper  = mount(<Loader {...props}/>);
        expect(wrapper.find(Fade).props().children.props.children[1].props.children).toEqual('Processing ...');
    })
})