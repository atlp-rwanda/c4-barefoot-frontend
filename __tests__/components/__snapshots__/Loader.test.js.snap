// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should be rendered without errors 1`] = `
<Fragment>
  <ForwardRef(Modal)
    className="makeStyles-modal-1"
    closeAfterTransition={true}
    open={true}
  >
    <ForwardRef(Fade)
      in={true}
    >
      <div
        className="makeStyles-loader-2"
      >
        <WithStyles(ForwardRef(CircularProgress)) />
        <WithStyles(ForwardRef(Typography))>
          Processing ...
        </WithStyles(ForwardRef(Typography))>
      </div>
    </ForwardRef(Fade)>
  </ForwardRef(Modal)>
</Fragment>
`;
